functions/generate_data_parallel.py

```diff
@@ -886,13 +886,19 @@ def apply_correction_io(local_dir,
                 corrected_experiment_data = limma.removeBatchEffect(
                     experiment_data, batch=experiment_map)
 
+                # Convert R object to pandas df
```

Comment:
Do you need to convert from R object to pandas dataframe twice? Does ComBat take in a dataframe and output an object?

Reply:
Nevermind, my bad. These are in two separate blocks. That being said, it's worth considering pulling the object conversion out of the if/else statement. It's helpful to have only the things that are directly different in if/else statements, to make it clear what actually changes in the process by choosing different parameters and what stays the same.